packages/ai/src/methods/chat-session.test.ts

```diff
@@ -156,5 +221,161 @@ describe('ChatSession', () => {
       );
       clock.restore();
     });
+    it('logs error and rejects user promise when response aggregation fails', async () => {
+      const loggerStub = stub(logger, 'error');
+      const error = new Error('Aggregation failed');
+
+      // Simulate stream returning, but the response promise failing (e.g. parsing error)
+      stub(generateContentMethods, 'generateContentStream').resolves({
+        stream: (async function* () {})(),
+        response: Promise.reject(error)
+      } as unknown as GenerateContentStreamResult);
+
+      const chatSession = new ChatSession(fakeApiSettings, 'a-model');
+      const initialHistoryLength = (await chatSession.getHistory()).length;
+
+      // Immediate call resolves with the stream object
+      const result = await chatSession.sendMessageStream('hello');
+
+      // User's response promise should reject
+      await expect(result.response).to.be.rejectedWith(error);
+
+      // Wait for the internal _sendPromise chain to settle
+      await new Promise(resolve => setTimeout(resolve, 0));
+
+      expect(loggerStub).to.have.been.calledWith(error);
+
+      // History should NOT have been updated (no response appended)
+      const finalHistory = await chatSession.getHistory();
+      expect(finalHistory.length).to.equal(initialHistoryLength);
+    });
+    it('logs error but resolves user promise when history appending logic fails', async () => {
+      const loggerStub = stub(logger, 'error');
+
+      // Simulate a response that is technically valid enough to resolve aggregation,
+      // but malformed in a way that causes the history update logic to throw.
+      // Passing `null` as a candidate causes `{ ...response.candidates[0].content }` to throw.
```

Comment:
This is a good test and we should keep it, but surfaces that we should probably fix the logic to protect against response.candidates[0].content throwing. Can you add a todo for me? In chat-session.ts line 188.